iroh-relay/src/server/resolver.rs

```diff
@@ -0,0 +1,90 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::{anyhow, Result};
+use reloadable_state::Reloadable;
+use rustls::{
+    server::{ClientHello, ResolvesServerCert},
+    sign::CertifiedKey,
+};
+use tokio_util::task::AbortOnDropHandle;
+
+/// A Certificate resolver that reloads the certificate every interval
+#[derive(Debug)]
+pub struct ReloadingResolver<Loader: Send + 'static> {
+    /// The inner reloadable value.
+    reloadable: Arc<Reloadable<CertifiedKey, Loader>>,
+    /// The handle to the task that reloads the certificate.
+    _handle: AbortOnDropHandle<()>,
+    /// Shutdown signal sender
+    _shutdown_tx: tokio::sync::oneshot::Sender<()>,
```

Comment:
Should I bubble this up into the relay so we call it on relay.shutdown() or is the AbortOnDropHandle enough for this usecase?